### DISCUSSION TITLE
How to know the list of all apps on which Monterey is not supported anymore?

### DISCUSSION THREAD

**Discussion by anta40** [Homebrew/discussions#5615] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.23
ORIGIN: https://github.com/Homebrew/brew
HEAD: 42c11a4dedf352229271593de831161071603c66
Last commit: 24 hours ago
Core tap JSON: 18 Sep 03:11 UTC
Core cask tap JSON: 18 Sep 03:11 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.x0XOFZVmjG/org.xquartz:0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 3.3.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: quad-core 64-bit broadwell
Clang: 14.0.0 build 1400
Git: 2.46.1 => /usr/local/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 12.7.6-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  phantomjs

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  python-platformdirs
  python-distlib
  python-filelock
  ciphey

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  ciphey
  neofetch
  openssl@1.1
  python-pytz
  python-typing-extensions
  pyyaml
  simh
  smali
  spotify-tui

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-versions
  Homebrew/homebrew-cask-fonts
Untap them with `brew untap`.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libmpeparser64.0.dylib
  /usr/local/lib/libpas2jslib.dylib
  /usr/local/lib/vfxsupp64.1.dylib

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  fpc
  numpy
  pillow
  mgba
  far2l

Warning: You are using macOS 12.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

I'm on macOS Monterey 12.7.6, and just noticed some maintainers stopped providing support for this OS, e.g [flyctl](https://formulae.brew.sh/formula/flyctl), [pandoc](https://formulae.brew.sh/formula/pandoc), and [virtualenv](https://formulae.brew.sh/formula/virtualenv). It's not the end of the world (at least for now), just a liitle bit annoying because Homebrew builds the apps from source instead of installing the binaries.

Now I'm curious, is there a quick way to know all apps on which Monterey isn't supported anymore?

**Comment by ZhongRuoyu** [comment, marked as answer]
> I'm on macOS Monterey 12.7.6, and just noticed some maintainers stopped providing support for this OS, e.g flyctl, pandoc, and virtualenv.

A note here: Homebrew's maintainership model is different from some other packagers'; maintainers do not own or exclusively maintain a package. Instead, macOS Monterey support has been dropped for all new builds from last week onwards, for the reason you see in your `brew doctor` output.

> Now I'm curious, is there a quick way to know all apps on which Monterey isn't supported anymore?

By "not supported" I assume you mean the unavailability of bottles (prebuilt binaries). You can try

```
HOMEBREW_DEVELOPER=1 brew unbottled --tag=monterey `brew list`
```

For arm64, change the tag to `arm64_monterey`. To see all unbottled formulae instead of the ones installed on your system, substitute `` `brew list` `` with `--eval-all`. For more information, see [manpage](https://docs.brew.sh/Manpage#unbottled-options-formula-).

**Reply by konnano** [reply]
I am using OS 10.13
I made it by myself out of the same thought

https://github.com/konnano/brew_list

**Reply by anta40** [reply]
> A note here: Homebrew's maintainership model is different from some other packagers'; maintainers do not own or exclusively maintain a package. Instead, macOS Monterey support has been dropped for all new builds from last week onwards, for the reason you see in your `brew doctor` output.

Hmmm I don't get this. For example:

- [Pandoc](https://formulae.brew.sh/formula/pandoc) doesn't support Monterey anymore. Now the minimum OS is Ventura.
- [GCC](https://formulae.brew.sh/formula/gcc) still supports Monterey.
- [FPC](https://formulae.brew.sh/formula/fpc) sill supports Big Sur.

So why there are differences?

> By "not supported" I assume you mean the unavailability of bottles (prebuilt binaries). You can try

Yes. Perhaps I need to make myself clear: bottles which are not available on the entire online Homebrew repo (not the ones installed on my Mac)

**Reply by carlocab** [reply]
> So why there are differences?

We always build bottles for all supported systems every time a formula has been updated. (Note however that sometimes a bottle just won't build on a specific system, but those are exceptions.) When a new OS is released, we build new bottles for the new OS but don't discard the old ones.

So, this just means:
- `pandoc` has been updated since we dropped Monterey support
- `gcc` hasn't been updated since then
- `fpc` was last updated while we still supported Big Sur

**Reply by ZhongRuoyu** [reply]
> So why there are differences?

This is because `gcc` / `fpc` hasn't been updated since we dropped support for Monterey / Big Sur. We don't actively remove bottles for old macOS releases; bottles for old releases only get dropped when a new build is needed -- typically when there is a new release -- because we don't build on the old platforms anymore.
